src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -667,4 +675,49 @@ private String getFileName(File f) {
             return fileName;
         }
     }
+
+    /**
+     * issues #1111
+     * Check the recipe of deleted component, if it uses docker image and the image not used by other, delete the image.
+     *
+     * @param componentIdentifier  identifier for the component
+     * @throws PackageLoadingException this may from findComponentRecipeContent
+     */
+    private void removeDockerImage(ComponentIdentifier componentIdentifier) throws PackageLoadingException {
+        logger.atDebug("removeDockerImage-start").log();
+        List<String> allRecipeContentList = componentStore.getAllComponentRecipeContent();
+        Optional<String> componentRecipeContent = componentStore.findComponentRecipeContent(componentIdentifier);
+        List<String> toBeRemovedList = new ArrayList<String>();
+        if (componentRecipeContent.isPresent()) {
+            RecipeLoader.RecipeFormat format = RecipeLoader.RecipeFormat.YAML;
+            // the recipe is JSON or YML
+            if (componentRecipeContent.get().startsWith("{")) {
+                format = RecipeLoader.RecipeFormat.JSON;
+            }
+            // get recipe
+            com.amazon.aws.iot.greengrass.component.common.ComponentRecipe componentRecipe =
+                    RecipeLoader.parseRecipe(componentRecipeContent.get(), format);
+            List<String> currentUriList =
+                    componentRecipe.getManifests().stream()
+                            .map(PlatformSpecificManifest::getArtifacts)
+                            .flatMap(Collection::stream)
+                            .map(item -> item.getUri().toString())
+                            .collect(Collectors.toList());
+            // if starts with "docker:", which means this uri use docker image
+            currentUriList = currentUriList.stream()
+                    .filter(item -> item.startsWith("docker:")).collect(Collectors.toList());
+            for (String uriString : currentUriList) {
+                // this docker image only used by itself
+                if (allRecipeContentList.stream().filter(item -> item.contains(uriString)).count() < 2) {
+                    toBeRemovedList.add(uriString);
+                }
+            }
+            // run docker command to remove image
+            Exec exec = Platform.getInstance().createNewProcessRunner();
+            logger.info("removeDockerImage-start" + toBeRemovedList.toString());
+            toBeRemovedList.forEach(uriString -> {
```

Comment:
Please implement this [here](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DefaultDockerClient.java) along with the rest of the docker cli usage  
And use the [parser](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageArtifactParser.java) to get the image information from the artifact URI